bc/subscription/api_views.py

```diff
@@ -61,6 +62,10 @@ def handle_cl_webhook(request: Request) -> Response:
                 timedelta(seconds=settings.WEBHOOK_DELAY_TIME),
                 process_filing_webhook_event,
                 filing.pk,
+                retry=Retry(
+                    max=settings.RQ_MAX_NUMBER_OF_RETRIES,
+                    interval=settings.RQ_RETRY_INTERVAL,
+                ),
```

Comment:
You're right about the main task, We don't have much logic that could fail in there but I added the retry object because We have a request (`lookup_documents_by_id`) there that could fail due to a time-out and We should ask RQ to retry the task.

If The main task fails because the request raised an exception, RQ will retry this task after a few seconds ( We can set the interval using the variable named `RQ_RETRY_INTERVAL`) but It won't create duplicate posts in the channels because the main task crashes before the code reaches the loop to schedule the subtasks. ( This PR also fixes https://github.com/freelawproject/bigcases2/issues/164)

On the other hand, if a subtask fails It will only retry itself because the main task is not waiting until the subtask finishes, the main task just enqueues the subtasks and finish its execution. It works like this:

1. The main task starts.
2. The main task links the webhook to a subscription, checks the webhook description, and tries to retrieve the document ( if something fails, RQ will retry this task).
3. The main task enters the loop and starts scheduling the subtasks.
4. the main task finishes.
5. The Twitter subtask sends a tweet, then the mastodon subtask fails ( the mastodon subtask will get a retry)
6. The mastodon subtask tries to create a new post and fails again. 
7. The mastodon subtask tries again.
8. ...

Let me know what you think.

Reply:
Sounds good, thanks.